### PR TITLE
Basic bundling

### PR DESCRIPTION
**Pull request by ankrgyl**

### CODE REVIEW

File: core/js/typespecs/app_types.ts [hunk 1]
```diff
@@ -173,39 +173,84 @@ export const datasetSchema = z
   .openapi("Dataset");
 export type Dataset = z.infer<typeof datasetSchema>;
 
+export const validRuntimes = ["node"] as const;
```
Comment (manugoyal) [inline core/js/typespecs/app_types.ts:176]: Would prefer to define the enums as zod enums, similar to how we do elsewhere in this file. Just

```
export const runtimesSchema = z.enum(["node"]);
export type Runtime = z.infer<typeof runtimeSchema>
```

You can iterate through the list of members with `runtimesSchema.options`.

File: core/js/typespecs/app_types.ts [hunk 2]
```diff
@@ -173,39 +173,84 @@ export const datasetSchema = z
...
+      z.strictObject({ score: z.number() }),
+    ]),
+  }),
+  bundle_id: z.string(),
+});
+export type CodeBundle = z.infer<typeof codeBundleSchema>;
+
+export const functionDataSchema = z.union([
+  z.strictObject({
+    type: z.literal("prompt"),
```
Comment (manugoyal) [inline core/js/typespecs/app_types.ts:235]: Maybe a comment stating that for backwards-compatibility reasons, the prompt data is hoisted outside this object.

File: js/src/logger.ts [hunk 3]
```diff
@@ -2714,6 +2714,10 @@ export class ReadonlyExperiment extends ObjectFetcher<ExperimentEvent> {
 
 let executionCounter = 0;
 
+export function newId() {
```
Comment (manugoyal) [inline js/src/logger.ts:2717]: What's the point of this?

File: js/src/framework.ts [hunk 4]
```diff
@@ -233,8 +234,14 @@ function initExperiment<IsOpen extends boolean = false>(
...
 }
 
+export type SpanContext = {
+  currentSpan: typeof currentSpan;
+  NOOP_SPAN: typeof NOOP_SPAN;
+};
+
 declare global {
   var _evals: EvaluatorFile;
+  var _spanContext: SpanContext | undefined;
```
Comment (manugoyal) [inline js/src/framework.ts:244]: Where is this used?

**Review by manugoyal — APPROVED** [review]

File: js/src/logger.ts (hunk 3 above)
Comment (ankrgyl) [reply js/src/logger.ts:2717]: Just factor this logic out so I don't need to remember how we create a uuid each place we need to create an id.

File: core/js/typespecs/app_types.ts (hunk 1 above)
Comment (ankrgyl) [reply core/js/typespecs/app_types.ts:176]: can do

**Pull request merged**